tex2pdf_service/tex2pdf/remote_call.py

```diff
@@ -0,0 +1,73 @@
+"""Compile a tarball via the service URL to an outcome file."""
+
+import json
+import logging
+import os
+import tarfile
+import time
+
+import requests
+
+
+def get_outcome_meta(outcome_file: str) -> dict:
+    """Open a compressed outcome tar archive and get the metadata."""
+    meta = {}
+    with tarfile.open(outcome_file, "r:gz") as outcome:
+        for name in outcome.getnames():
+            if name.startswith("outcome-") and name.endswith(".json"):
+                meta_contents = outcome.extractfile(name)
+                if meta_contents:
+                    meta.update(json.load(meta_contents))
+    return meta
+
+
+def submit_tarball(service: str, tarball: str, outcome_file: str, tex2pdf_timeout: int, post_timeout: int, auto_detect: bool = False) -> bool:
+    """Submit tarball to compilation service."""
```

Comment:
[Naming discussion]

The `submit` verb is a bit loaded due to the arXiv "submission" process, which gets me a little confused. If you're only thinking of message passing here, how about `send_tarball`?

But I see that the function does the web request and handles the response, so this is the "full work". Maybe then `process_tarball` is even better?

Reply:
I changed it to `service_process_tarball`, since `process_tarball` itself again has the same problem ;-)